src/core/service.c

```diff
@@ -1941,8 +1941,6 @@ static bool service_will_restart(Unit *u) {
 
```

Comment:
The commit message of the first commit is bit outdated?
I guess "This essentially reverts 2ad2e41a72ec19159c0746a78e15ff880fe32a63." should be moved to the second commit??

Reply:
Hmm, I was referring to the `UNIT_NOTIFY_WILL_AUTO_RESTART` logic 🤔

Reply:
That makes sense. Updated.